### PR TITLE
Use mypy for type checking

### LINKED ISSUE #244: Use MyPy
To improve the static robustness of this library (and its usages), we should integrate MyPy.

### PR DESCRIPTION
**Pull request by rschwanhold**
## Description
fixes #244 

## Todo
- [x] Add typechecking to CI

### CODE REVIEW

File: wkcuber/convert_nifti.py [hunk 1]
```diff
@@ -111,18 +113,17 @@ def to_target_datatype(
...
-    layer_name,
-    dtype,
-    scale,
-    mag=1,
-    file_len=DEFAULT_WKW_FILE_LEN,
-    bbox_to_enforce=None,
-    write_tiff=False,
-    use_orientation_header=False,
-    flip_axes=None,
-):
```
Comment (rschwanhold) [inline wkcuber/convert_nifti.py:125]: The parameter `mag` is never used (it gets overwritten). Therefore, I removed the parameter. Is that correct?

File: wkcuber/api/Properties/LayerProperties.py [hunk 2]
```diff
@@ -161,23 +176,23 @@ def bounding_box(self) -> dict:
...
-        mappings=None,
-    ):
+        name: str,
+        category: str,
+        element_class: str,
+        data_format: str,
+        num_channels: int,
+        bounding_box: Dict[str, Union[int, Tuple[int, int, int]]] = None,
+        resolutions: List[Resolution] = None,
+        largest_segment_id: int = None,
```
Comment (rschwanhold) [inline wkcuber/api/Properties/LayerProperties.py:193]: I think `largest_segment_id` is not optional for segmentation layers, but making it non-optional would require a different order of the parameters and then the order of `SegmentationLayerProperties` would be different from `LayerProperties`. 
(There is an assertion in DatasetProperties._add_layer for this)
Therefore, I would leave it this way.

**Review by rschwanhold — COMMENTED** [review]
I had do add a few exceptions to handle the `None` case of `Optional`. Let me know if there is a better way to do this

**Review by jstriebel — APPROVED** [review]
Nice! I did some spot-checks, they LGTM :+1: I guess `mypy` does the rest ;-) It's awesome that you already checked this with the current mypy-repo, so we don't expect any new errors when doing https://github.com/scalableminds/voxelytics/issues/1982 here, too.

File: wkcuber/convert_nifti.py (hunk 1 above)
Comment (jstriebel) [reply wkcuber/convert_nifti.py:125]: Seems fine for me, let's just check who might call this in any dependent projects.

File: wkcuber/api/Properties/LayerProperties.py (hunk 2 above)
Comment (jstriebel) [reply wkcuber/api/Properties/LayerProperties.py:193]: Yes, I think you can leave it like this or have a default of `-1`, if that is useful? If you want to leave the signature as is, you can `assert largest_segment_id is not None` in the `__init__()` to mark the attribute as non-optional.

**Comment by jstriebel** [comment]
> I had do add a few exceptions to handle the None case of Optional. Let me know if there is a better way to do this.

We often use assertions in voxelytics for that matter, e.g. `assert x is not None, "x must be set to an integer in case ABC"`.

**Comment by philippotto** [comment]
Are there any blockers for this PR or can it be merged soon? :)

**Comment by rschwanhold** [comment]
This will be merged today

**Pull request merged**